viewing_party/party.py

```diff
@@ -1,23 +1,147 @@
 # ------------- WAVE 1 --------------------
+from tests.test_constants import MOVIE_TITLE_1, RATING_1
 
+#make a dictioary with three parameters if three of them are truthy
+#if any of them are none function returns none
 def create_movie(title, genre, rating):
-    pass
+    
 
-# -----------------------------------------
+    new_movie=dict([("title",title),("genre", genre),("rating",rating)])
+  
+    
+    if title is None:
+        return None
+    if genre is None:
+        return None
+    if rating is None:
+        return None
+    return new_movie
+
+#This function needs to add movie to the watched list inside the user_data
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+#This function needs to add movie to the watchlist inside the user_data
+def add_to_watchlist(user_data,movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+#If title in watchlist remove and add to watched. Else return user data as is.
+def watch_movie(user_data,title):
+    for i in user_data["watchlist"]:
+        if i["title"]==title:
+            user_data["watched"].append(i)
+            user_data["watchlist"].remove(i)
+    return user_data
+    # 
+# ----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+#calculate the average rating of user watched list. if watched is empty return 0.
+def get_watched_avg_rating(user_data):
+    test_list=user_data["watched"]
+    ress =  [i["rating"] for i in test_list]
+    if len(ress)==0:
+        return 0
+    else:
+        return sum(ress)/len(ress)
 
+#  Determine which genre is most frequently occurring in the watched list. Return that. if watched is empty return None.
+def get_most_watched_genre(user_data):
+    User_watched_movies=user_data["watched"]
+    Users_watched_genre =  [i["genre"] for i in User_watched_movies]
+    frequency={}
+    if len(User_watched_movies)>0:
 
+        for item in Users_watched_genre:
+            if item in frequency:
+                frequency[item] += 1
+            else:
+                frequency[item] = 1
+        fav_genre = max(frequency, key=frequency.get)
+        return fav_genre
+    else:
+        return None

```

Comment:
I'd suggest flipping the order of the conditions around to handle the special case (ie, checking if something is falsy) first, so that the code can quickly check and get out of there. This arrangement is referred to as a "guard clause" or a "guard condition." Notice that it lets us move the main focus of the function (creating a dictionary from the parameters) out of an indented block, letting it stand out more clearly.

```suggestion
    if len(User_watched_movies)<0:
        return None
    frequency={}
    for movie in Users_watched_genre:
        if item in frequency:
            frequency[item] += 1
        else:
            frequency[item] = 1
    fav_genre = max(frequency, key=frequency.get)
    return fav_genre
```